testament/specs.nim

```diff
@@ -18,7 +18,17 @@ type TestamentData* = ref object
 
 let testamentData0* = TestamentData()
 
-var compilerPrefix* = findExe("nim")
+# First check the bin directory and then check the system path.
+let
+  nimBinPath = findExe("bin/nim")
+  nimTempPath = findExe("bin/nim_temp")
```

Comment:
having fallbacks like that leads to subtle bugs where you expect it picks up one thing but instead of failing loudly when the 1st path is invalid, it silently accepts the fallback.

instead, use the existing `--nim:bin/nim_temp`, no magic path needed

Reply:
So we don't need `nimBinPath` and no fallbacks? I think If testament is a exposed test tool, It should check the default Nim path and #15013 is not accepted?

Reply:
my comment was regarding `bin/nim_temp`; whether the default is nim in PATH or bin/nim, I'm fine with either, because `--nim:` can override anyways

Reply:
Thanks. I'll do it.

Reply:
Maybe we need something like `testament --temp cat /` because `koch temp` could be used heavily.

Reply:
> having fallbacks like that leads to subtle bugs where you expect it picks up one thing but instead of failing loudly when the 1st path is invalid, it silently accepts the fallback.

Exactly!